server/blob/blob-s3/src/test/java/org/apache/james/blob/objectstorage/aws/S3DeDuplicationBlobStoreTest.java

```diff
@@ -37,18 +37,24 @@
 
     @BeforeAll
     static void setUpClass(DockerAwsS3Container dockerAwsS3) {
-        AwsS3AuthConfiguration configuration = AwsS3AuthConfiguration.builder()
+        AwsS3AuthConfiguration authConfiguration = AwsS3AuthConfiguration.builder()
             .endpoint(dockerAwsS3.getEndpoint())
             .accessKeyId(DockerAwsS3Container.ACCESS_KEY_ID)
             .secretKey(DockerAwsS3Container.SECRET_ACCESS_KEY)
             .build();
 
-        s3BlobStoreDAO = new S3BlobStoreDAO(configuration, dockerAwsS3.dockerAwsS3().region());
+        S3BlobStoreConfiguration s3Configuration = S3BlobStoreConfiguration.builder()
+            .authConfiguration(authConfiguration)
+            .region(dockerAwsS3.dockerAwsS3().region())
+            .build();
+
+        s3BlobStoreDAO = new S3BlobStoreDAO(s3Configuration);
+
         testee = BlobStoreFactory.builder()
             .blobStoreDAO(s3BlobStoreDAO)
             .blobIdFactory(new HashBlobId.Factory())
             .defaultBucketName()
-            .passthrough();
+            .deduplication();
```

Comment:
Were we playing dedup test suite against a passthrough blobstore by mistake?

Could it be the route cause of your troubles?

Reply:
No I found the root cause, nothing to do with this. That's just little fix, was testimng a passthrough in dedup tests and dedup in passthrough tests. Technically it doesn't change except matching correctly the test suite name